crates/matrix-sdk-indexeddb/src/event_cache_store/transaction.rs

```diff
@@ -393,4 +393,184 @@ impl<'a> IndexeddbEventCacheStoreTransaction<'a> {
     {
         self.transaction.object_store(T::OBJECT_STORE)?.clear()?.await.map_err(Into::into)
     }
+
+    /// Query IndexedDB for chunks that match the given chunk identifier in the
+    /// given room. If more than one item is found, an error is returned.
+    pub async fn get_chunk_by_id(
+        &self,
+        room_id: &RoomId,
+        chunk_id: &ChunkIdentifier,
+    ) -> Result<Option<Chunk>, IndexeddbEventCacheStoreTransactionError> {
+        self.get_item_by_key_components::<Chunk, IndexedChunkIdKey>(room_id, chunk_id).await
+    }
+
+    /// Add a chunk to the given room and ensure that the next and previous
+    /// chunks are properly linked to the chunk being added. If a chunk with
+    /// the same identifier already exists, the given chunk will be
+    /// rejected.
+    pub async fn add_chunk(
+        &self,
+        room_id: &RoomId,
+        chunk: &Chunk,
+    ) -> Result<(), IndexeddbEventCacheStoreTransactionError> {
+        self.add_item(room_id, chunk).await?;
+        if let Some(previous) = chunk.previous {
+            let previous_identifier = ChunkIdentifier::new(previous);
+            if let Some(mut previous_chunk) =
+                self.get_chunk_by_id(room_id, &previous_identifier).await?
+            {
+                previous_chunk.next = Some(chunk.identifier);
+                self.put_item(room_id, &previous_chunk).await?;
+            }
+        }
+        if let Some(next) = chunk.next {
+            let next_identifier = ChunkIdentifier::new(next);
+            if let Some(mut next_chunk) = self.get_chunk_by_id(room_id, &next_identifier).await? {
+                next_chunk.previous = Some(chunk.identifier);
+                self.put_item(room_id, &next_chunk).await?;
```

Comment:
`put_item` confuses me a lot. Shouldn't be renamed to `update_item`?

Reply:
So, `put_item` is named after the functions exposed by [`indexed_db_futures`](https://docs.rs/indexed_db_futures/0.5.0/indexed_db_futures/idb_object_store/struct.IdbObjectStore.html#method.put_val), which are named after the functions exposed in [JavaScript IndexedDB API](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/put).

That being said, I'm happy to rename my wrapper functions using `update` language, if you prefer! I was just trying to be consistent with the existing API.

Reply:
Okay, it's justified, all good.